### ISSUE TITLE
Why changing is_active is disabled in email confirmation?

### ISSUE THREAD

**Issue by bnisevic** [pennersr/django-allauth#1408]
And what you find as proper way to change is_active value when email is confirmed by clicking on the link with key in it?

I did the view override. Is that proper way?

https://github.com/pennersr/django-allauth/blob/master/allauth/account/views.py#L243-L248

**Comment by pennersr** [comment]
As per the Django documentation on `is_active`:

"Boolean. Designates whether this user account should be considered active. We recommend that you set this flag to False instead of deleting accounts; that way, if your applications have any foreign keys to users, the foreign keys won’t break."

So using it for email verification is wrong. It is also often used to blacklist users. In any case, allauth does not touch it. There is no need to. Whether or not an email is verified is kept track of via the `EmailAddress` model.

**Comment by bnisevic** [comment]
I understand your point of view.

But for me common sense would be:
1. register (signup) account
2. get verification email at registered email address
3. click on email sent with a key
4. click sets user to is_active = True (so far it was False)
5. And now part which you mentioned from Django is_active. If an admin wants to deactivate account he just sets is_active = False. The same can be used on user self deactivation/account removal.

Buy what are you telling when it comes to allauth, is_active will always be False.

**Comment by pennersr** [comment]
It's not set to `False`. `is_active` is not touched by allauth, so it will be set to the default value as defined in Django's `AbstractUser` (which is `True`). It is up to the project to decide what the exact semantics of `is_active` is. You might want to use it for marking accounts as deleted, you might want to use to indicate a user has a paid account. 

Having said this, you can easily achieve what you want by hooking up to some of the signals provided. 

I still do not think this is a good idea though. You are mixing two concepts: manual (de)activation by an admin, and email verification. So you have one field, `is_active`, which is mutated on the one hand automatically as part of your onboarding process, and on the other hand it is manually mutated by admins. So, whenever you see it having the value `False` it is not clear what happened. Did the admin deactivate the account because of abuse, or was the email address not verified?

**Comment by bnisevic** [comment]
OK. I got you. Thank you very much!

**Comment by JasonSpine** [comment]
I use Django 1.10.5, I wrote:
```
def activate(request, activationCode):
	user = User.objects.filter(last_name=activationCode)
	if user.exists():
		user[0].is_active = True
		print user[0].is_active
		user[0].save()
		print user[0].is_active
```
both prints give me False :(

**Comment by bnisevic** [comment]
I use CBVs in Django. And I inherited ConfirmEmailView from allauth where I overrided post method to be able to change is_active value.
I don't know what did you do to have last_name as activation code. But try to get user which you are activating from the account_emailaddress object.

**Comment by JasonSpine** [comment]
I didn't want to modify default User model at the moment, so I used last_name field to store activation key for the user. I know it's silly...

Now I modified User model:
```
from django.contrib.auth.models import AbstractUser

class MyUser(AbstractUser):
	activated = models.BooleanField(default=False)
	activationCode = models.CharField(max_length=40, default="")
```
And I had the same problem, couldn't change activated value. But then I tried to log the user and change activated value - it worked then!
It seems that Django doesn't allow changing user record if the user isn't logged.
So I modified my activation approach. I've added hidden activation code field to login form field. That field is filled with activation link GET data. Once user posts in his correct login data, he gets authenticated, activationCode from user record is compared to hidden form field and if there is a match, activated bool is set to true and user is saved correctly.
